components/registration-framework/org.wso2.carbon.identity.user.registration.engine/src/main/java/org/wso2/carbon/identity/user/registration/engine/graph/TaskExecutionNode.java

```diff
@@ -146,4 +155,31 @@ private Response handleCompleteStatus(RegistrationContext context, ExecutorRespo
         }
         return new Response.Builder().status(STATUS_COMPLETE).build();
     }
+
+    private Map<String, String> updateContextWithAuthenticatorProperties(RegistrationContext context,
```

Comment:
This method fetch the default authenticator properties of a given idp and return those as a map.
Can we update the method name to match the purpose (getDefaultAuthenticatorConfigsByIDPName). Also let's update the method signature to accept tenant domain and idp name.

Reply:
Another option is to populate the the RegistrationContext inside the method itself and make it void.